x-pack/plugins/security_solution/public/cases/components/user_action_tree/user_action_copy_link.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback } from 'react';
+import { EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import { useParams } from 'react-router-dom';
+import copy from 'copy-to-clipboard';
+
+import { useGetUrlSearch } from '../../../common/components/navigation/use_get_url_search';
+import { SecurityPageName } from '../../../app/types';
+import { navTabs } from '../../../app/home/home_navigations';
+import * as i18n from './translations';
+
+interface UserActionCopyLinkProps {
+  id: string;
+}
+
+const UserActionCopyLinkComponent = ({ id }: UserActionCopyLinkProps) => {
+  const { detailName: caseId } = useParams<{ detailName: string }>();
+  const urlSearch = useGetUrlSearch(navTabs.case);
+
+  const handleAnchorLink = useCallback(() => {
```

Comment:
This is not on you more on me, but it will be nicer to use this hook to build the url `useFormatUrl`; right now I think we might have a bug with it.